site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java

```diff
@@ -9749,215 +9762,262 @@ private void actionForTemplate(String direction, int index,
-							try {
-								Site sourceSite = SiteService.getSite(oldSiteId);
-								sourceSiteRef = sourceSite.getReference();
-								
-							} catch (IdUnusedException e) {
-								log.warn(this + ".actionForTemplate; case29: invalid source siteId: "+oldSiteId);
-								return;
+							String newSiteId = null;
+							if (StringUtils.trimToNull(params.getString("newSiteId")) == null) {
+								newSiteId = IdManager.createUuid();
+							} else{
+								newSiteId = params.getString("newSiteId");
 							}
 
-							// SAK-20797
-							long oldSiteQuota = this.getSiteSpecificQuota(oldSiteId);
-
-							Site site = SiteService.addSite(newSiteId,
-									getStateSite(state));
-
-							boolean removeStealthToolsFromDup = ServerConfigurationService.getBoolean(SAK_PROP_RM_STLTH_ON_DUP, SAK_PROP_RM_STLTH_ON_DUP_DEFAULT);
-							if (removeStealthToolsFromDup) {
-								List<SitePage> pageList = site.getPages();
-								if (CollectionUtils.isNotEmpty(pageList)) {
-									List<SitePage> rmPageList = new ArrayList<>();
+							try {
+								String oldSiteId = (String) state
+										.getAttribute(STATE_SITE_INSTANCE_ID);
 
-									// Check if each tool is stealthed; if so, queue for removal
-									for (SitePage page : pageList) {
-										List<ToolConfiguration> pageToolList = page.getTools();
-										if (CollectionUtils.isNotEmpty(pageToolList)) {
-											List<ToolConfiguration> rmToolList = new ArrayList<>();
+								// Retrieve the source site reference to be used in the EventTrackingService
+								// notification of the start/end of a site duplication.
+								String sourceSiteRef = null;
+								try {
+									Site sourceSite = SiteService.getSite(oldSiteId);
+									sourceSiteRef = sourceSite.getReference();
 
-											for (ToolConfiguration toolConf : pageToolList) {
-												Tool tool = toolConf.getTool();
-												String toolId = StringUtils.trimToEmpty(tool.getId());
+								} catch (IdUnusedException e) {
+									log.warn(this + ".actionForTemplate; case29: invalid source siteId: "+oldSiteId);
+									return;
+								}
 
-												if (StringUtils.isNotBlank(toolId) && !notStealthOrHiddenTool(toolId)) {
-													// Found a stealthed tool, queue for removal
-													log.debug("found stealthed tool {}", toolId);
-													rmToolList.add(toolConf);
+								// SAK-20797
+								long oldSiteQuota = this.getSiteSpecificQuota(oldSiteId);
+
+								// Create the duplicate site
+								Site site = SiteService.addSite(newSiteId, getStateSite(state));
+								site.setTitle(titleStripped);
+
+								// If the site contains the Email Archive tool, we're going to check for valid/unique email address;
+								// this requires the duplicate site to already exist...
+								if (state.getAttribute(STATE_DUP_SITE_HAS_EMAIL_ARCHIVE) != null) {
+									String newEmailID = StringUtils.trimToNull(params.getString("emailAddress"));
+									if (StringUtils.isBlank(newEmailID)) {
```

Comment:
already trimmed to null directly above

Reply:
ditto here, this is existing code that was just moved around.